File: core/src/nl/t64/game/rpg/screens/world/entity/Direction.java
package nl.t64.game.rpg.screens.world.entity;

import com.badlogic.gdx.math.MathUtils;


public enum Direction {
    NORTH,
    SOUTH,
    WEST,
    EAST,

    NONE;

    public static Direction getRandom() {
        int randomNumber = MathUtils.random(getAllDirectionsWithoutNONE());
        return Direction.values()[randomNumber];
    }

    private static int getAllDirectionsWithoutNONE() {
        return Direction.values().length - 2;
    }

}

File: core/src/nl/t64/game/rpg/screens/world/entity/Direction.kt
package nl.t64.game.rpg.screens.world.entity

import com.badlogic.gdx.math.MathUtils


enum class Direction {
    NORTH,
    SOUTH,
    WEST,
    EAST,

    NONE;

    companion object {
        fun getRandom(): Direction {
            val randomNumber = MathUtils.random(getAllDirectionsWithoutNONE())
            return values()[randomNumber]
        }

        private fun getAllDirectionsWithoutNONE(): Int = values().size - 2
    }

}
